SwiftCompilerSources/Sources/Optimizer/FunctionPasses/ComputeSideEffects.swift

```diff
@@ -142,6 +142,15 @@ private struct CollectedEffects {
         handleApply(pa)
         checkedIfDeinitBarrier = true
       }
+      // Escaping closures capture arguments by moving them into the
+      // context. This has the effect of reading the captured values.
+      // Nonescaping closures may bitwise-move value into the context,
+      // but this has no effect.
+      if !pa.isOnStack {
+        for operand in pa.operands {
```

Comment:
We don't need to include the first operand (= the callee). For this you can write
```
  for argument in pa.arguments {
```

Reply:
@eeckstein I intended to handle the callee argument in case the callee is itself an escaping closure which might have been passed in. I need to add back a comment about that and come up with a SIL test case.